### DISCUSSION TITLE
Can GSLP be used to layout a document?

### DISCUSSION THREAD

**Discussion by mundacho** [eclipse-glsp/glsp#462] [Q&A] [answered]
Hello,

I was wondering if it's possible to use GLSP to create a document layout editor. For example, I would like to  have the client be able to customize an invoice by placing some lines of text on a page and inserting a logo and being able to put it the way they like. Is this kind of thing possible using GSLP? If not, do you know which framework can achieve this sort of thing?

**Comment by planger** [comment, marked as answer]
Hi Edmundo,

Interesting use case! In general this should work with GLSP imho, but is certainly not the common use case. So you'll probably need to customize some tools here and there.
One thing that e.g. will need customization is text word-wrapping. SVG doesn't support word-wrapping. So you'll need to use foreignObject, as used e.g. in this [Sprotty example](https://github.com/eclipse/sprotty/blob/ddf634cc877cb6169eb3de90cfc8a465c8e516c3/examples/svg/src/standalone.ts#L60).

**Reply by mundacho** [reply]
Thanks! I'll start by looking into that!
